dvc/repo/update.py

```diff
@@ -1,14 +1,20 @@
+from ..dvcfile import Dvcfile
 from . import locked
 
 
 @locked
-def update(self, target, rev=None):
-    from ..dvcfile import Dvcfile
+def update(self, targets=None, rev=None, recursive=False):
+    if not targets:
+        stages = self.collect(targets, recursive=recursive)
+    else:
+        stages = set()
+        for target in targets:
```

Comment:
Could also
```
if isinstance(targets, str):
    targets = [targets]     
```
just as a convenience (we have that in `dvc add` too), since it is guaranteed someone will try to misuse it :slightly_smiling_face:

Reply:
By the way are the any other commands which should contain `isinstance(targets, str)` check? `repo.used_cache`, `repo.status`, `repo._checkout` seem good candidates.

Reply:
@nik123 Probably not `used_cache` but the rest could use it, sure. But let's not bother with it in this PR, it is out of scope :slightly_smiling_face: